pkg/cloudprovider/aws/factory.go

```diff
@@ -92,6 +92,12 @@ func (f *Factory) CapacityFor(provisioner *v1alpha1.Provisioner) cloudprovider.C
 	}
 }
 
+func (f *Factory) TerminateFor() cloudprovider.Termination {
+	return &Termination{
```

Comment:
Consider not even wrapping this struct. InstanceProvider already satisfies the terminate interface.

Reply:
Cool! Didn't know you could do that.